Create emergency_xeno shuttle

TheOnlySolitaire opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->
<!--
IMPORTANT: For large prs, make sure include in your title [IDB IGNORE] and/or [MDB IGNORE] to keep their bots from crashing. 
MDB is especially important for large map changes
Failure to do so could result in repoban
If you need help please reach out on the discord!
-->

## About The Pull Request

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

New shuttle based on an pre-existing shuttle that has been overrun by Xenos. 
The end-goal is to have the shuttle be unlocked by scanning a xenos organ and/or by purchasing the corresponding xeno research. 

## Why It's Good For The Game

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

New shuttle adds to options and fun in the game.

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl:
add: Adds new shuttle
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

SgtHunk commented:
How does this interact with shuttle-stalling antags, now that because of the shuttle we have 10+ facehuggers that will make player-controlled xenos

TheOnlySolitaire commented:
> How does this interact with shuttle-stalling antags, now that because of the shuttle we have 10+ facehuggers that will make player-controlled xenos

Very good point. I think Shuttle Hijack could be considerably harder, though a clever antag might use this shuttle to assist in preventing other crew getting onboard so they just need to focus on the hijacking and surviving part. But for sure preventing crew from escaping would become easier. 

Something to consider though is that there are the cult and doom arena shuttles, thats off the top of my head and there may be a couple more, which also have this issue. Is this that far different? 
Would balancing the number of possible player controlled mobs/facehuggers make it more agreeable?

StoneMason commented:
If there's a blob, revolution or even a smart cult, the shuttle will arrive and self-antaggers/dumbasses will sprint onto the shuttle and get facehugged. Whilst it probably wouldn't be a huge issue as captain's rarely go for cool shuttles, this shit will definitely cause a xeno outbreak a few times. Just make it like the lavaland ruin so it's xenomobs and popped eggs.

SgtHunk reviewed (changes requested):
Remove the xeno eggs entirely. A shuttle shouldn't be bringing an antag onto the station. It can include enemies a la cult shuttle, but having people get facehugged and bring xenos is just too much.

wraith-54321 commented:
1 egg in a command only area could work, I dont know, or you could just give the xeno mobs a fun balloon like in the cult shuttle

TheOnlySolitaire commented:
I will remove the eggs and just leave mobs, is it ok for those mobs and at least for now, remove the sentience balloon to let them be player controlled. If thats ok and requested I'll re-add it but eggs will be gone on next commit.

Helianthus00 commented:
This seems a little odd to just have available to order at any time. If you lock it behind something like emag or xeno research, I'd be more willing to consider this.

JohnFulpWillard commented on fulp_modules/mapping/shuttles/map_templates.dm:
```diff
@@ -45,3 +45,14 @@
 	description = "The standard-issue escape shuttle for models of station this large. Will get you home in moderate style."
+	
+/datum/map_template/shuttle/emergency/fulp/xeno
+	prefix = "fulp_modules/mapping/shuttles/evac/"
```
delete this line

TheOnlySolitaire commented:
> This seems a little odd to just have available to order at any time. If you lock it behind something like emag or xeno research, I'd be more willing to consider this.

It was always planned to be, I just needed to figure it out. I've updated the file so that it now has:

- locked the shuttle behind alien tech
- removed Queen and Praetorian
- removed all xeno eggs
- removed sentience balloon
- tweaked xeno mobs so there are 7 total, 3 drones, 3 sentinels and 1 non-hostile larva
